storage/client/queue/src/main/java/com/azure/storage/queue/QueueServiceClientBuilder.java

```diff
@@ -230,11 +233,28 @@ public QueueServiceClientBuilder credential(SharedKeyCredential credential) {
      * @throws NullPointerException If {@code connectionString} is {@code null}.
      */
     public QueueServiceClientBuilder connectionString(String connectionString) {
-        Objects.requireNonNull(connectionString);
         this.sharedKeyCredential = SharedKeyCredential.fromConnectionString(connectionString);
+        getEndPointFromConnectionString(connectionString);
         return this;
     }
 
+    private void getEndPointFromConnectionString(String connectionString) {
+        HashMap<String, String> connectionStringPieces = new HashMap<>();
+        for (String connectionStringPiece : connectionString.split(";")) {
```

Comment:
Since you just need the account name, do you need to look at all the connection string key-value pairs?

Reply:
Alan has the refactor in his common module PR.
https://github.com/alzimmermsft/azure-sdk-for-java/blob/AzStorage_Common_Package/storage/client/queue/src/main/java/com/azure/storage/queue/QueueClientBuilder.java#L284